gluu-server/opt/gluu/jetty/oxauth/custom/scripts/person_authentication/SignInCanada.py

```diff
@@ -222,13 +223,16 @@ def prepareForStep(self, configurationAttributes, requestParameters, step):
                     if mfaId is None:
                         print("%s: prepareForStep. mfaId is missing!" % self.name)
                         return False
-                    loginHint = "%s|%s" % (mfaId, entityId)
-                    passportOptions["login_hint"] = Base64Util.base64urlencode(crypto.encryptAES(self.aesKey, loginHint))
+                    passportOptions["login_hint"] = mfaId
+                    # The following parameters are redundant, but currently required by the 2ndFaaS
+                    passportOptions["redirect_uri"] = self.passport.getProvider(provider)["callbackUrl"]
+                    passportOptions["response_type"] = "code"
+                    passportOptions["scope"] = "openid profile"
```

Comment:
Adding profile dynamically to the scope. So please ignore my first comment above.